src/bplus_tree.rs

```diff
@@ -0,0 +1,525 @@
+use crate::chunk_pointer::{ChunkHandler, ChunkPointer};
+use std::{
+    fs::File,
+    io::{self, ErrorKind},
+    os::unix::fs::FileExt,
+    path::PathBuf,
+};
+extern crate chunkfs;
+use chunkfs::{Data, DataContainer, Database};
+
+const MAXSIZE: u64 = 2 << 20;
+
+pub struct BPlus<K> {
+    t: usize,
+    root: Option<Box<Node<K>>>,
+    path: PathBuf,
+    file_number: usize,
+    offset: u64,
+    current_file: File,
+}
+
+#[derive(Clone)]
+#[allow(dead_code)]
+struct Node<K> {
+    leaf: bool,
+    key_num: usize,
+    keys: Vec<K>,
+    children: Vec<Option<Box<Node<K>>>>,
+    pointers: Vec<ChunkHandler>,
+}
+
+impl<K: Ord + Clone + Default> Node<K> {
+    fn new(leaf: bool) -> Self {
+        let new_node = Node {
+            leaf: leaf,
+            key_num: 0,
+            keys: Vec::new(),
+            children: Vec::new(),
+            pointers: Vec::new(),
+        };
+        new_node
+    }
+}
+#[allow(dead_code)]
+impl<K: Ord + Clone + Default> BPlus<K> {
+    pub fn new(t: usize, path: PathBuf) -> Self {
+        let root = Node::new(true);
+        let mut nodes: Vec<Box<Node<K>>> = Vec::new();
+        nodes.push(Box::new(root));
+        let path_to_file = path.join("0");
+        let res = File::create(path_to_file);
+        let current_file;
+        match res {
+            Ok(x) => current_file = x,
+            Err(_x) => panic!(),
```

Comment:
```suggestion
        let current_file = File::create(path_to_file)?;
```
Нужно будет тогда еще поменять возвращаемый тип на io::Result<Self>

Если хочешь именно панику, можно просто сделать unwrap